plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/errors/VerificationError.kt

```diff
@@ -45,4 +47,19 @@ object ErrorAdapter {
             else -> UnknownError(err)
         }
     }
-}
\ No newline at end of file
+}
+
+/**
+ * Given a verification error, find embedded extra information of type `I` in the
+ * error's offending nodes.
+ * If the extra info is not found, return `null`.
+ * The information can be embedded either in the result's offending node,
+ * or in the reason's offending node.
+ * As an example, `PreconditionInCallFalse` errors have
+ * as offending node result the call-site of the called method.
+ * But the actual info we are interested in is on the pre-condition, contained in the reason's offending node.
+ */
+inline fun <reified I> VerificationError.getInfoOrNull(): I? =
+    Info.fromSilver(result.offendingNode().prettyMetadata._2()).unwrapOr<I> {
+        Info.fromSilver(result.reason().offendingNode().prettyMetadata._2()).unwrapOr<I> { null }
```

Comment:
`.prettyMetadata._2()` probably deserves to be a helper function/property with a meaningful name.